cmd/promlts/sidecar.go

```diff
@@ -93,7 +116,56 @@ func runSidecar(
 			s.Stop()
 			l.Close()
 		})
+	}
+	// Periodically query the Prometheus config. We use this as a heartbeat as well as for updating
+	// the external labels we apply.
+	{
+		promUp := prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "promlts_sidecar_prometheus_up",
+			Help: "Boolean indicator whether the sidecar can reach its Prometheus peer.",
+		})
+		lastHeartbeat := prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "promlts_sidecar_last_heartbeat_seconds",
```

Comment:
Hm.. Can we name it differently? My first guess after seeing `promlts_sidecar_last_heartbeat_seconds` is how many seconds passed from last heartbeat (which does not make sense in gauge) - not the exact timestamp.. maybe just make it a counter and  name `promlts_sidecar_heartbeats_count`

Reply:
Having time information on when we actually last contacted it can be quite valuable – but probably then we only want to see the last successful timestamp.

Maybe adjust those semantics and name it `promlts_sidecar_last_heartbeat_success_time_seconds`?

Reply:
+!